addons/website_sale_coupon/controllers/main.py

```diff
@@ -16,6 +17,26 @@ def pricelist(self, promo, **post):
             request.session['error_promo_code'] = coupon_status['error']
         return request.redirect(post.get('r', '/shop/cart'))
 
+    @http.route(['/shop/cart/update_json'], type='json', auth="public", methods=['POST'], website=True, csrf=False)
+    def cart_update_json(self, product_id, line_id=None, add_qty=None, set_qty=None, display=True, **kw):
+        if line_id and product_id and set_qty == 0:
+            product = request.env['product.product'].browse(int(product_id))
+            line = request.env['sale.order.line'].sudo().browse(int(line_id))
+            if line.is_reward_line:
+                status = request.session.get('pending_coupon') or {}
+                code = status.get('promo_code')
+
+                pending_coupon = request.env['coupon.coupon'].sudo().search([('code', '=', code)], limit=1)
```

Comment:
only need to apply it if a pending_coupon is in session...
you are doing useless query

Reply:
Do you think it's better to check `is_reward_line` first (triggers a read) or to check the pending coupon first?

Reply:
> only need to apply it if a pending_coupon is in session...

Check session first